### PR TITLE
NH-3486 - Performance: For projections, identical metadata for rows is recalculated for every row

### PR DESCRIPTION
**Pull request by lnu**
related to https://github.com/nhibernate/nhibernate-core/pull/446
Changes done in CriteriaLoader instead of CriteriaQueryTranslator.

### CODE REVIEW

File: src/NHibernate/Loader/Criteria/CriteriaLoader.cs [hunk 1]
```diff
@@ -48,6 +51,8 @@ public class CriteriaLoader : OuterJoinLoader
 			resultTypes = walker.ResultTypes;
 			includeInResultRow = walker.IncludeInResultRow;
 			resultRowLength = ArrayHelper.CountTrue(IncludeInResultRow);
+			cachedProjectedTypes = translator.ProjectedTypes;
```
Comment (hazzik) [inline src/NHibernate/Loader/Criteria/CriteriaLoader.cs]: this will be the same as `resultTypes`

**Comment by hazzik** [comment]
I get a lot of test failing

**Comment by lnu** [comment]
Ok i'll check it but I ran everything locally and it seemed ok. 
thanks

**Comment by hazzik** [comment]
@lnu you need `if(translator.HasProjection)` around variable assignments

**Comment by lnu** [comment]
I've just run it again and you're right :+1:

File: src/NHibernate/Loader/Criteria/CriteriaLoader.cs (hunk 1 above)
Comment (lnu) [inline src/NHibernate/Loader/Criteria/CriteriaLoader.cs]: @hazzik I can safely use resultTypes instead of cachedProjectedTypes?

Comment (hazzik) [inline src/NHibernate/Loader/Criteria/CriteriaLoader.cs]: @lnu yes, it just set in walker using the same `translator.HasProjection` condition

**Pull request merged**